### PR TITLE
Document remaining OS functions

### PR DESCRIPTION
**Pull request by Lemmmy**
This documents the remaining functions in the OS API and adds a stub for `_G`. Also clarified the function of timer and alarm IDs.

### CODE REVIEW

File: doc/stub/os.lua [hunk 1]
```diff
@@ -1,6 +1,121 @@
...
+return `false`. If you want to handle this more gracefully, use an alternative
+such as @{loadfile}.
+
+@tparam table env The environment to run the program with.
+@tparam string path The exact path of the program to run.
+@param ... The arguments to pass to the program.
+@treturn boolean Whether or not the program ran successfully.
+@usage Run the default shell from within your program:
+
+os.run({}, "/rom/programs/shell")
```
Comment (SquidDev) [inline doc/stub/os.lua:116]: I think this needs to be indented 4 spaces (or use code blocks?).

File: src/main/java/dan200/computercraft/core/apis/OSAPI.java [hunk 2]
```diff
@@ -199,11 +205,14 @@ public final void cancelTimer( int token )
...
     /**
      * Sets an alarm that will fire at the specified world time. When it fires,
-     * an alarm event will be added to the event queue.
+     * an {@code alarm} event will be added to the event queue with the ID
+     * returned from this function as the first parameter.
      *
      * @param time The time at which to fire the alarm, in the range [0.0, 24.0).
-     * @return The ID of the alarm that was set.
+     * @return The ID of the new alarm. This can be used to filter the
+     *   {@code alarm} event, or @{os.cancelAlarm|cancel the alarm}.
```
Comment (SquidDev) [inline src/main/java/dan200/computercraft/core/apis/OSAPI.java:213]:
```suggestion
     *   {@code alarm} event, or {@link #cancelAlarm cancel the alarm}.
```

Haven't tested, but _should_ work.

File: src/main/java/dan200/computercraft/core/apis/OSAPI.java [hunk 3]
```diff
@@ -171,12 +171,18 @@ public final void queueEvent( String name, IArguments args )
...
+     * the ID returned from this function as the first parameter.
+     *
+     * As with @{os.sleep|sleep}, {@code timer} will automatically be rounded up
+     * to the nearest multiple of 0.05 seconds, as it waits for a fixed amount
+     * of world ticks.
      *
      * @param timer The number of seconds until the timer fires.
-     * @return The ID of the new timer.
+     * @return The ID of the new timer. This can be used to filter the
+     *   {@code timer} event, or @{os.cancelTimer|cancel the timer}.
```
Comment (SquidDev) [inline src/main/java/dan200/computercraft/core/apis/OSAPI.java:183]:
```suggestion
     *   {@code timer} event, or {@link #cancelTimer cancel the timer}.
```
Haven't tested, but _should_ work. Won't work with `sleep` for obvious reasons.

**Review by SquidDev — CHANGES_REQUESTED** [review]
Generally looks good, thank you! Just left a couple of nitpicks.

File: doc/stub/os.lua (hunk 1 above)
Comment (Lemmmy) [reply doc/stub/os.lua:116]: Whoops, got inlined when turning the comment into --[[

**Review by SquidDev — APPROVED** [review]

**Comment by SquidDev** [comment]
Thank you!

**Pull request merged**